src/test/java/com/amazon/ion/bytecode/bin11/bytearray/ListOpcodeHandlerTests.kt

```diff
@@ -0,0 +1,725 @@
+                Instructions.I_LIST_START.packInstructionData(9),
+                Instructions.I_LIST_START.packInstructionData(7),
+                Instructions.I_LIST_START.packInstructionData(5),
+                Instructions.I_LIST_START.packInstructionData(3),
+                Instructions.I_LIST_START.packInstructionData(1),
+                *Array(16) { Instructions.I_END_CONTAINER }.toIntArray()
+            )
+            ShortLengthPrefixedListOpcodeHandler.shouldCompile(bytes, expectedBytecode)
+        }
+    }
+
+    /*
+     * ================================================
+     * ==            LONG PREFIXED LISTS             ==
+     * ================================================
+     */
+
+    @Nested
+    inner class `long length-prefixed lists` {
+        @Test
+        fun `handler compiles lists of simple elements`() {
+            val bytes = "FA 0D 61 01 61 02 61 03".hexStringToByteArray()
+            val expectedBytecode = intArrayOf(
+                Instructions.I_LIST_START.packInstructionData(4),
+                Instructions.I_INT_I16.packInstructionData(1),
+                Instructions.I_INT_I16.packInstructionData(2),
+                Instructions.I_INT_I16.packInstructionData(3),
+                Instructions.I_END_CONTAINER
+            )
+            LongLengthPrefixedListOpcodeHandler.shouldCompile(bytes, expectedBytecode)
+        }
+
+        @Test
+        fun `handler compiles empty lists`() {
+            val bytes = "FA 01".hexStringToByteArray()
+            val expectedBytecode = intArrayOf(
+                Instructions.I_LIST_START.packInstructionData(1),
+                Instructions.I_END_CONTAINER
+            )
+            LongLengthPrefixedListOpcodeHandler.shouldCompile(bytes, expectedBytecode)
+        }
+
+        @Test
+        fun `handler compiles large lists`() {
+            val testLength = 10_000_000 // Much larger causes test crash
+            val bytes = byteArrayOf(
+                OpCode.VARIABLE_LENGTH_LIST.toByte(),
+                *generateFlexUIntBytes(testLength),
+                *Array<Byte>(testLength) { 0x6E /* true */ }.toByteArray()
+            )
+            val expectedBytecode = intArrayOf(
+                Instructions.I_LIST_START.packInstructionData(testLength + 1),
+                *Array(testLength) { Instructions.I_BOOL.packInstructionData(1) }.toIntArray(),
+                Instructions.I_END_CONTAINER
+            )
+            LongLengthPrefixedListOpcodeHandler.shouldCompile(bytes, expectedBytecode)
+        }
+
+        @Test
+        fun `handler compiles nested lists`() {
+            val bytes = """
+                FA 63
+                FA 11                        | [null, [[null, null]], null]
+                8E                           |--|     |               |
+                FA 09 FA 05 8E 8E            |--------|               |
+                8E                           |------------------------|
```

Comment:
The lines were a bit confusing at first. Maybe something like this instead?
```suggestion
                FA 11                        | [
                8E                           |   null,
                FA 09 FA 05 8E 8E            |   [[null, null]],
                8E                           |   null,
                                             | ]
```